### PR TITLE
Emit object:upload:finished event when object upload finishes

### PR DESCRIPTION
**Pull request by Nicktho**
me: can u upload this pls
satay: 📣 "finished uploading file a" 📣
satay: 📣 "finished uploading file b" 📣
me: ty

### CODE REVIEW

File: src/api/uploadObjectsToBucket.ts [hunk 1]
```diff
@@ -21,7 +35,12 @@ export async function uploadObjectsToBucket(
           Key: keyOfObjectToUpload,
           Body: fs.createReadStream(statsOfObjectToUpload.path)
         })
-        .promise();
+        .promise()) as SendDataWithVersion;
+      emitter.emit('object:upload', keyOfObjectToUpload, {
+        progress: 100,
+        version: VersionId,
+        status: ObjectDiffStatus.ADDED
```
Comment (jameslnewell) [inline src/api/uploadObjectsToBucket.ts]: status could be `Modified` or `Added` - maybe just remove `ObjectEvent .status` for now to get this merged

Comment (Nicktho) [reply src/api/uploadObjectsToBucket.ts]: true! and also i was thinking with progess: 0 does `Modified` or `Added` make sense? I'll go ahead and take it out for now

Comment (jameslnewell) [reply src/api/uploadObjectsToBucket.ts]: the tense `ed` definitely doesn't make sense at `0` but whether its being added or modified still makes some sense.

`version` is a little weird at `0` too... oh well, that's what breaking changes are for.

File: src/api/deleteObjectsFromBucket.ts [hunk 2]
```diff
@@ -20,4 +28,10 @@ export async function deleteObjectsFromBucket(
       }
     })
     .promise();
+  Deleted.forEach(({Key, VersionId}) => {
```
Comment (jameslnewell) [inline src/api/deleteObjectsFromBucket.ts]: does this return the version deleted?

**Pull request merged**